lib/sift/where_handler.rb

```diff
@@ -19,16 +19,13 @@ def apply_jsonb_conditions(collection, value)
 
       value.each do |key, val|
         collection = if val.is_a?(Array)
-          if val.include?(nil)
-            condition = val.each_with_index.map do |element, i|
-              "#{@param.internal_name}->>'#{key}' #{element === nil ? 'IS NULL' : "= :value_#{i}"}"
-            end.join(' OR ')
-            elements = Hash[val.each_with_index.map { |item, i| ["value_#{i}".to_sym, item.to_s] } ]
-          else
-            condition =  "('{' || TRANSLATE(#{@param.internal_name}->>'#{key}', '[]','') || '}')::text[] && ARRAY[?]"
-            elements = val.map(&:to_s)
-          end
-          collection.where(condition, elements)
+          elements = Hash[val.each_with_index.map { |item, i| ["value_#{i}".to_sym, item.to_s] } ]
+          elements[:all_values] = val.compact.map(&:to_s)
+          main_condition =  "('{' || TRANSLATE(#{@param.internal_name}->>'#{key}', '[]','') || '}')::text[] && ARRAY[:all_values]"
+          sub_conditions = val.each_with_index.map do |element, i|
+            "#{@param.internal_name}->>'#{key}' #{element === nil ? 'IS NULL' : "= :value_#{i}"}"
```

Comment:
Does it make sense to subcondition every value since they would be handled by inclusion in the main_condition array? Maybe there should just be a single subcondition for when element === nil.

Reply:
The values could be inside an array: `{"items" => [1,13,25] }` (main condition) but also could be stored directly: `{ "items" => false }` (sub_conditions)